### DISCUSSION TITLE
about native code plugins :  what c# attribute maps to  required 'directive'  when generating  json schema to be sent to llm

### DISCUSSION THREAD

**Discussion by sabbadino** [microsoft/semantic-kernel#8457] [Q&A] [answered]
I cannot find documentation on this

**Comment by evchaki** [comment]
@sabbadino - can you provide some additional context here on what are you trying to do?  Sounds like you are doing function calling here?

**Reply by sabbadino** [reply]
Yes sure , tools/ function calling .. I would like to have a clear documentation about the mapping between
 c# code (of course attributes included ) defining function and its parameters to -> Json schema sent to LLM (nullability, required , etc .. )
Thanks

**Comment by markwallace-microsoft** [comment, marked as answer]
Hi @sabbadino 

currently we only have a Blog post on this topic for Python https://devblogs.microsoft.com/semantic-kernel/diving-into-function-calling-and-its-json-schema-in-semantic-kernel-python/

I am going to work on a C# equivalent during our next sprint (start next week). Here is the tracking item: https://github.com/microsoft/semantic-kernel/issues/8534.

**Reply by sabbadino** [reply]
Ok thanks
